### ISSUE TITLE
Improvement: also allow UTF-16 for prints

### ISSUE THREAD

**Issue by toonsevrin** [EOSIO/eos#3247]
This is a very low-prio issue but it would be nice to see UTF-16 support for the prints function that contracts call to log a string to console.

Converting to UTF-8 is a bit costly for languages that use UTF-16 internally, native support for both formats would thus be nice.

**Comment by tbfleming** [comment]
Action data and database data are also UTF-8. UTF-16 is wasteful in constrained environments. Maybe the assemblyscript library for contracts could have a UTF-8 string type?

**Comment by toonsevrin** [comment]
Exactly, that's why I marked this as low-priority. The guys over at assemblyscript have already hacked together a fast toUTF8 encoder, which definitely gets the job done.

**Comment by taokayan** [comment]
UTF-16 is not backward compatible with ANSI string. It possibly breaks the existing code and consume 100% more space.
